src/lib.rs

```diff
@@ -4,6 +4,12 @@ use std::os::unix::io::{AsRawFd, RawFd};
 use std::os::unix::net;
 use std::{alloc, io, mem, ptr};
 
+#[cfg(feature = "tokio_0_2")]
+extern crate tokio_0_2 as tokio;
+
+#[cfg(feature = "tokio_0_3")]
+extern crate tokio_0_3 as tokio;
```

Comment:
renaming both `as tokio` makes features not additive due to conflicting names. With both enabled you should end up with impls for both tokio 0.2 and 0.3.

Reply:
Right, the alternative is to duplicate the implementation code, do we want that?

Reply:
It is already fairly deduplicated by the virtue of those impls just calling a single function, isn't it? A macro could reduce LOC further. Overall I don't think saving on LOC in exchange for surprising behaviour of the features is worth it.

Reply:
Done